Sprint-3/todo-list/02-guide_to_modularize_code.md

```diff
@@ -0,0 +1,40 @@
+# Why modularize code?
+
+Modularizing code means breaking a program into smaller, self-contained pieces (modules), each responsible for a specific functionality. This approach offers several key advantages:
+
+- Reusability – You can use the same code in different parts of a project or in other projects.
+- Maintainability – It's easier to fix bugs or update features when code is organized into smaller parts.
+- Separation of Concerns – Each part of the code handles a specific task, keeping things clear and organized.
+- Team Collaboration – Multiple people can work on different modules at the same time without conflict.
+- Scalability – You can add new features more easily as the project grows.
+- Testing – Smaller modules are easier to test individually.
```

Comment:
It feels maybe worth giving an "all together" vs "split out" example alongside this content? I think "separation of concerns" and "testability" are quite hard things to visualise without an example if you haven't experienced them before.